File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    int answer = 1; int paint=section[0]+m;
    for(int i=1; i<section.size(); i++){
        if(paint > section[i]){
            continue;
        }
        else{
            answer++;
            paint=section[i]+m;
        }
    }
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 1;
        int paint = section[0]+m;
        for(int i=1; i<section.length; i++){
            if(paint>section[i]){
                continue;
            }
            else{
                answer++;
                paint=section[i]+m;
            }
        }
        return answer;
    }
}